app/controllers/projects_controller.rb

```diff
@@ -135,6 +149,68 @@ def export_list_modal
 
   private
 
+  def from_template? = @template.present?
+
+  def new_blank
+    @new_project = @parent&.children&.build || Project.new
+
+    render layout: "no_menu"
+  end
+
+  def new_from_template
+    @copy_options = Projects::CopyOptions.new
+    @new_project = Projects::CopyService
+      .new(user: current_user, source: @template, contract_options: { validate_model: false })
+      .call(target_project_params: params.permit(:parent_id).to_h, attributes_only: true)
+      .result
+
+    render layout: "no_menu"
+  end
+
+  def create_blank
+    service_call = Projects::CreateService
+      .new(user: current_user)
+      .call(permitted_params.project)
+
+    @new_project = service_call.result
+
+    if service_call.success?
+      redirect_to project_path(@new_project), notice: I18n.t(:notice_successful_create)
+    else
+      flash[:error] = I18n.t(:notice_unsuccessful_create_with_reason, reason: service_call.message)
+      render action: :new, status: :unprocessable_entity
+    end
+  end
+
+  def create_from_template # rubocop:disable Metrics/AbcSize
+    @copy_options = Projects::CopyOptions.new(copy_options_params)
+
+    service_call = Projects::EnqueueCopyService
+      .new(user: current_user, model: @template)
+      .call(
+        target_project_params: permitted_params.project.to_h,
+        only: @copy_options.dependencies,
+        send_notifications: @copy_options.send_notifications
+      )
+
+    if service_call.success?
+      job = service_call.result
+      redirect_to job_status_path(job.job_id)
+    else
+      @new_project = service_call.result
+      flash[:error] = I18n.t(:notice_unsuccessful_create_with_reason, reason: service_call.message)
+      render action: :new, status: :unprocessable_entity
+    end
+  end
+
+  def find_optional_template
+    @template = Project.visible(current_user).find(params[:template_id]) if params[:template_id].present?
```

Comment:
This still allows me to send in a project id that is not marked as being a template. The autocompleter does not show it, but we should not rely on the frontend being the only entity controlling this.
```suggestion
    @template = Project.visible(current_user).find_by(id: params[:template_id], templated: true) if params[:template_id].present?
```

Reply:
@klaustopher Yes, you're absolutely right. This is now fixed, albeit with a newly-introduced `templated` scope and `find`, rather than `find_by`.

N.B. `find`/`find_by` is probably preferable in this case (or at least for the `create` action) since passing a `template_id` param for a template the current user does not have permissions to view will result in a 404 being rendered - as opposed to attempting to create a blank project.